pyatv/protocols/mrp/__init__.py

```diff
@@ -502,6 +505,58 @@ async def artwork(
         return artwork
 
     async def _fetch_artwork(self, width, height) -> Optional[ArtworkInfo]:
+        return await self._fetch_remote_artwork(
+            width, height
+        ) or await self._fetch_local_artwork(width, height)
+
+    async def _fetch_remote_artwork(self, width, height) -> Optional[ArtworkInfo]:
+        """Fetch external artwork from a URL."""
+        metadata = self.psm.playing.metadata
+        if not metadata:
+            return None
+
+        urls = []
+
+        if metadata.HasField("artworkIdentifier"):
+            # appears to be a template to itunes artwork, but let's validate
+            url_template = metadata.artworkIdentifier
+            try:
+                url = url_template.format(
+                    # the itunes image server preserves aspect ratio
+                    w=999999 if width < 1 else width,
+                    h=999999 if height < 1 else height,
+                    c="bb",
+                    f="png",
+                )
+            except KeyError:
+                url = None
+            url_parts = urlparse(url)
+            if url_parts.scheme and url_parts.netloc:
+                urls.append(url)
+
+        if metadata.HasField("artworkURL"):
+            # artworkURL has fixed size and format, use it as a fallback
+            urls.append(metadata.artworkURL)
+
+        async with aiohttp.ClientSession() as session:
```

Comment:
You should not create a new session here, but rather re-use the one in the `pyatv.core.Core` instance (passed when instantiating the protocol). That session is re-used everywhere in pyatv and is usually provided by the user.